azure/services/scalesets/scalesets_test.go

```diff
@@ -53,46 +47,47 @@ func init() {
 	_ = clusterv1.AddToScheme(scheme.Scheme)
 }
 
-func TestNewService(t *testing.T) {
-	g := NewGomegaWithT(t)
-	scheme := runtime.NewScheme()
-	_ = clusterv1.AddToScheme(scheme)
-	_ = infrav1.AddToScheme(scheme)
-	_ = infrav1exp.AddToScheme(scheme)
-
-	cluster := &clusterv1.Cluster{
-		ObjectMeta: metav1.ObjectMeta{Name: "test-cluster"},
-	}
-	client := fake.NewClientBuilder().WithScheme(scheme).WithObjects(cluster).Build()
-	s, err := scope.NewClusterScope(context.Background(), scope.ClusterScopeParams{
-		AzureClients: scope.AzureClients{
-			Authorizer: autorest.NullAuthorizer{},
-		},
-		Client:  client,
-		Cluster: cluster,
-		AzureCluster: &infrav1.AzureCluster{
-			Spec: infrav1.AzureClusterSpec{
-				Location:       "test-location",
-				ResourceGroup:  "my-rg",
-				SubscriptionID: "123",
-				NetworkSpec: infrav1.NetworkSpec{
-					Vnet: infrav1.VnetSpec{Name: "my-vnet", ResourceGroup: "my-rg"},
-				},
-			},
-		},
-	})
-	g.Expect(err).NotTo(HaveOccurred())
-
-	mps, err := scope.NewMachinePoolScope(scope.MachinePoolScopeParams{
-		Client:           client,
-		MachinePool:      new(clusterv1exp.MachinePool),
-		AzureMachinePool: new(infrav1exp.AzureMachinePool),
-		ClusterScope:     s,
-	})
-	g.Expect(err).NotTo(HaveOccurred())
-	actual := NewService(mps, resourceskus.NewStaticCache(nil, ""))
-	g.Expect(actual).ToNot(BeNil())
-}
+// ToDo: Fix this test case
```

Comment:
@CecileRobertMichon @shysank -- Any suggestions on fixing this test case. This test case requires to import scope package and hence causes import loop.

Reply:
looking, I don't think we should be creating real scopes to test `NewService()`. `NewService` takes a ScaleSetScope, which we have a mock for, so we should be able to do:
```
mockCtrl := gomock.NewController(t)
defer mockCtrl.Finish()
 scopeMock := NewMockScaleSetScope(mockCtrl)
actual := NewService(scopeMock, resourceskus.NewStaticCache(nil, ""))
g.Expect(actual).ToNot(BeNil())
```

If that doesn't work, that test isn't testing much since the NewService func just sets the parameters into a struct and returns it, so worst case we can get rid of the test altogether.

Reply:
I tried something like below: 
```
scopeMock := mock_scalesets.NewMockScaleSetScope(mockCtrl)
	rec:=scopeMock.EXPECT()
	rec.SubscriptionID().Return("123")
	rec.BaseURI().Return("/dummy")
	rec.Authorizer().Return(scopeMock)
	actual := NewService(scopeMock, resourceskus.NewStaticCache(nil, ""))
	g.Expect(actual).ToNot(BeNil())
```

I added those mock `Returns` as they were expected but at the end I got error something like this:

```	
wrong type of argument 0 to Return for *mock_scalesets.MockScaleSetScope.Authorizer: *mock_scalesets.MockScaleSetScope is not assignable to autorest.Authorizer [/Users/kuashutosh/go/src/sigs.k8s.io/cluster-api-provider-azure/azure/services/scalesets/scalesets_test.go:66]
```